File: Homework/CSharp-Part-1/06. Loops/MMSA.cs
using System;

class MMSAOfNNumbers
{
    static void Main()
    {
        int numbersCount = int.Parse(Console.ReadLine());

        double min = int.MaxValue;
        double max = int.MinValue;
        double sum = 0;
        double avg = 0;

        int counter = 0;
        while (counter < numbersCount)
        {
            double nextNumber = double.Parse(Console.ReadLine());

            if (nextNumber > max)
            {
                max = nextNumber;
            }

            if (nextNumber < min)
            {
                min = nextNumber;
            }

            sum += nextNumber;
            counter++;
        }

        avg = sum / counter;

        Console.WriteLine("min={0:0.00}", min);
        Console.WriteLine("max={0:0.00}", max);
        Console.WriteLine("sum={0:0.00}", sum);
        Console.WriteLine("avg={0:0.00}", avg);
    }
}

File: Homework/JavaScript-Fundamentals/06. Loops/mmsa.js
function solve(args) {
    var min = Number.POSITIVE_INFINITY;
    var max = Number.NEGATIVE_INFINITY;
    var sum = 0;
    var avg = 0;

    var nextNumber;
    var currentIndex = 0;

    while (true) {
        nextNumber = +args[currentIndex++];

        if (nextNumber > max)
        {
            max = nextNumber;
        }

        if (nextNumber < min)
        {
            min = nextNumber;
        }

        sum += nextNumber;

        if (currentIndex == args.length) {
            break;
        }
    }

    avg = sum / args.length;

    console.log('min=' + min.toFixed(2));
    console.log('max=' + max.toFixed(2));
    console.log('sum=' + sum.toFixed(2));
    console.log('avg=' + avg.toFixed(2));
}